04-request-lifecycle/01-nestjs-components/interceptors/api-version.interceptor.ts

```diff
@@ -1,5 +1,18 @@
 import { NestInterceptor, ExecutionContext, CallHandler } from "@nestjs/common";
+import { tap, map } from "rxjs";
 
 export class ApiVersionInterceptor implements NestInterceptor {
-  intercept(context: ExecutionContext, next: CallHandler) {}
+  intercept(context: ExecutionContext, next: CallHandler) {
+    const currentDate = new Date();
+
+    return next.handle().pipe(
+      map((data) => {
+        return {
+          ...data,
+          apiVersion: "1.0",
+          executionTime: Number(new Date()) - Number(currentDate)
```

Comment:
здесь все отлично, только по заданию это поле должно быть строкой, так что можно сделать что-то вроде
```suggestion
          executionTime: `${Number(new Date()) - Number(currentDate)}ms`
```

Reply:
Спасибо, поправил